### DISCUSSION TITLE
Upgrading in production and using external services

### DISCUSSION THREAD

**Discussion by zeekrey** [saleor/saleor#8284] [Deployment] [answered]
Hey there,

I'm checking if Saleor is something to work with in our next ecommerce project and I got two questions:

**Is it possible to use an external postgres db?**
The docs say that it is recommended to use docker. Using the docker-compose script, the redis and postgres containers will run on the same machine as the core. Is it possible to run just the core and point to the external postgres db? Like https://planetscale.com/ for example? This leads to a second question:

**Upgrading saleor in production**
How I'm supposed to upgrade a running saleor instance? Is it possible to use the following flow or is this not possible?
`git pull -> docker build (only the core) -> push the image` Will this migrate for example the db schema on a external postgres db?

**Comment by patrys** [comment, marked as answer]
Yes, you can set the `DATABASE_URL` to any valid connection string. Then, of course, the Docker container needs to connect to that database, but it's okay for it to be an external resource (for example, an AWS RDS instance).

For upgrades, you will explicitly need to execute the migrate command inside one of your Docker containers. We can't run it automatically as multi-container deployments would then risk having multiple migrations running in parallel.
